src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java

```diff
@@ -43,24 +43,30 @@
     }
 
     @ApiModelProperty(position = 1, required = true)
+    @Valid
+    @EmbeddedId
+    public BaseAccessionVersionEntityId id;
+
+
+    @ApiModelProperty(position = 2, required = true)
     @NotNull
     @Size(min = 1, max = 255)
     @JsonProperty
-    @Id
+    @Column(nullable = false)
     private String hash;
 
-    @ApiModelProperty(position = 2, required = true)
+    @ApiModelProperty(position = 3, required = true)
```

Comment:
Do we really need the position to be declared? Every time we change the attributes these numbers need to be modified too.

Reply:
yes . Else the parameters is displayed by alphabetical order.

Reply:
And what is the problem with that? It's not like users are going to look into the code and be angry at the mismatch.